target_chains/ethereum/contracts/contracts/pulse/Pulse.sol

```diff
@@ -127,19 +128,41 @@ abstract contract Pulse is IPulse, PulseState {
             }
         }
 
-        // Parse price feeds first to measure gas usage
-        PythStructs.PriceFeed[] memory priceFeeds = IPyth(_state.pyth)
-            .parsePriceFeedUpdates(
-                updateData,
-                priceIds,
-                SafeCast.toUint64(req.publishTime),
-                SafeCast.toUint64(req.publishTime)
-            );
+        // TODO: should this use parsePriceFeedUpdatesUnique? also, do we need to add 1 to maxPublishTime?
```

Comment:
Note that the tests mock the parsePriceFeedUpdates call, so we don't actually know if it's being invoked correctly.

Reply:
I think using `parsePriceFeedUpdatesUnique` here is a good idea due to its deterministic behavior